### PR TITLE
Extend toml default-configs test

### PR DESCRIPTION
**Pull request by cryptocode**

### CODE REVIEW

File: nano/core_test/toml.cpp [hunk 1]
```diff
@@ -76,24 +76,162 @@ TEST (toml, daemon_config_update_array)
...
+	ASSERT_EQ (conf.port, defaults.port);
+
+	ASSERT_EQ (conf.rpc_process.io_threads, defaults.rpc_process.io_threads);
+	ASSERT_EQ (conf.rpc_process.ipc_address, defaults.rpc_process.ipc_address);
+	ASSERT_EQ (conf.rpc_process.ipc_port, defaults.rpc_process.ipc_port);
+	ASSERT_EQ (conf.rpc_process.num_ipc_connections, defaults.rpc_process.num_ipc_connections);
+}
+
 /** Empty config file should match a default config object */
 TEST (toml, daemon_config_deserialize_defaults)
```
Comment (guilhermelawless) [inline nano/core_test/toml.cpp]: I think `conf_height_processor_batch_min_time` is missing

File: nano/core_test/toml.cpp [hunk 2]
```diff
@@ -76,24 +76,162 @@ TEST (toml, daemon_config_update_array)
 	ASSERT_EQ (c.node.preconfigured_peers[0], "test-peer.org");
 }
 
+/** Empty rpc config file should match a default config object */
+TEST (toml, rpc_config_deserialize_defaults)
+{
+	std::stringstream ss;
+
+	// A config file with values that differs from test-net defaults
```
Comment (guilhermelawless) [inline nano/core_test/toml.cpp]: This comment is out of place

File: nano/core_test/toml.cpp [hunk 3]
```diff
@@ -76,24 +76,162 @@ TEST (toml, daemon_config_update_array)
...
+
+	ASSERT_EQ (conf.node.websocket_config.enabled, defaults.node.websocket_config.enabled);
+	ASSERT_EQ (conf.node.websocket_config.address, defaults.node.websocket_config.address);
+	ASSERT_EQ (conf.node.websocket_config.port, defaults.node.websocket_config.port);
+
+	ASSERT_EQ (conf.node.callback_address, defaults.node.callback_address);
+	ASSERT_EQ (conf.node.callback_port, defaults.node.callback_port);
+	ASSERT_EQ (conf.node.callback_target, defaults.node.callback_target);
+
+	ASSERT_EQ (conf.node.ipc_config.transport_domain.allow_unsafe, defaults.node.ipc_config.transport_domain.allow_unsafe);
```
[the remark this answers is not in the document]
Comment (cryptocode) [reply nano/core_test/toml.cpp]: ah yeah, io_threads for ipc was special under json, i.e. never written unless non-default. Should work fine under the new toml framework so will change.

File: nano/core_test/toml.cpp (hunk 1 above)
Comment (cryptocode) [reply nano/core_test/toml.cpp]: yeah https://github.com/nanocurrency/nano-node/pull/2105 was missed wrt toml.

Comment (cryptocode) [reply nano/core_test/toml.cpp]: #2105 was missed

**Pull request merged**